Microsoft/botbuilder-js
# Add support for channelData field in conversation references

## Description

The conversationReference type is missing fields necessary to use `createConversation` with MS Teams.  This PR adds those fields and amends the functions so that they will work with Teams.

## Specific Changes

Adds the channelData field to the conversation reference.  Information in channelData may be required for operations that use references -- for example, createConversation.

Amends the adapter's createConversation method to pass through the tenant id, which is necessary for using this method with MS Teams.

## Testing

* Connect a bot to Teams.
* Using a conversation reference extracted from an incoming GROUP message, use adapter.createConversation to create a new PRIVATE context.
* Use the new context to send a private message to the user.  Note that it is received as a DM, instead of a reply in the group channel.
